components/VM_Arm/src/main.c

```diff
@@ -865,7 +873,10 @@ static int load_vm(vm_t *vm, const char *kernel_name, const char *dtb_name, cons
         }
     }
 
-    if (!config_set(CONFIG_VM_DTB_FILE)) {
+    ZF_LOGW_IF(vm_image_config.provide_dtb && vm_image_config.generate_dtb,
```

Comment:
How about getting rid of both these booleans and reserving some magic string for autogeneration. For example, setting `dtb_name` to `<generate>` is not a likely filename since it would create havoc in build systems due to `<` and `>` being shell redirections. The second case is leaving `dtb_name` empty and no DTB is loaded. Any other string should be considered a filename.

Reply:
Actually, there are multiple options, see also my comment https://github.com/seL4/camkes-vm/pull/48#discussion_r1007410679, so using your proposal we'd have:
- don't provide DTB. Could be `dtb_name = ""`
- provide a fixed TB from `dtb_name`. Do not modify it. Could be `dtb_name = "some.dtb"`
- generate as DTB from scratch. Could be `dtb_name = "<generate>"`
- generate as DTB based on a give DTB. Should this be handled via `dtb_name = "<generate>some.dtb"`

Reply:
How about:

- Don't provide DTB: `dtb_name = ""`
- Provide fixed DTB: `dtb_name = "some.dtb"`
- Generate from scratch: `dtb_name = "<"`
- Generate from given DTB: `dtb_name = "<some.dtb"`

Note dropped `>` at the end. Now you can just check if first character is `<`, if yes, use the remaining string (which might be empty) as filename. No need for error prone string copies. Any other character than `<` would be fine to me, as long as it is unlikely to appear in real world filenames.

Reply:
That looks nice actually. We just have to clearly document this somewhere then also.